parallelize run_DomainAnnotationSets

jmchandonia opened the pull request:
this passed the unit test when run on docker03

dcchivian reviewed (changes requested):
need to update RELEASE_NOTES.md to report changes

dcchivian reviewed (approved):
looks good.  We should both test in a Narrative.  What I don't know is where is best to test it, since we need a worker pool and maybe that's only on PROD?  I'll ask on Slack.  Let me know by email how your tests go.

jmchandonia commented:
command line unit test works.  fixed bugs identified by lgtm.  added line to release notes.  haven't registered yet in ci or beta; let me know what you hear from slack on where to test it.
